### PR TITLE
[Backend] Update Main Lambda to Format Emails and Add Attachments

### PR DESCRIPTION
**Pull request by mtruong-dbx**
Here are some screenshots of the files in Dropbox!
Screen Shot 2021-07-14 at 9 42 48 PM
Screen Shot 2021-07-14 at 9 42 59 PM
Screen Shot 2021-07-14 at 9 43 05 PM

### CODE REVIEW

File: backend/dropbox_client.py [hunk 1]
```diff
@@ -6,24 +6,35 @@ def __init__(self):
             oauth2_access_token="temp_token"
         )
 
-    def upload_file_to_dropbox(self, email_contents, path, desired_file_name, access_token):
-        content_in_bytes = self.convert_email_content_to_bytes(email_contents)
-        path = "%s/%s" % (path, desired_file_name)
-
+    def upload_file_to_dropbox(self, file_contents, path, file_name, access_token):
```
Comment (mtruong-dbx) [inline backend/dropbox_client.py:9]: Note: Most of this code was copied over from https://github.com/rogebrd/mailman/pull/14

File: backend/formatted_email.py [hunk 2]
```diff
@@ -2,39 +2,36 @@
...
     keyword_line_indices = [i for i in range(len(lines)) if FROM_KEYWORD in lines[i]]
     cut_index = keyword_line_indices[0] if keyword_line_indices else 0
     formatted_text = "\n".join(lines[cut_index:])
     self.formatted_text = formatted_text
-  
+
   def set_foldername(self, lines):
     if not lines:
-      return
+      return self.formatted_subject
```
Comment (mtruong-dbx) [inline backend/formatted_email.py:27]: Note: I think we should revisit how we want to name the folders in Dropbox. We can fix this in a follow up diff

File: backend/mailman_lambda.py [hunk 3]
```diff
@@ -27,7 +28,9 @@ def lambda_handler(event, context):
             )
             item = response['Item']
             oauth_token = item['dropbox_oauth_token']['S']
-            dropbox_client.upload_file_to_dropbox(email.text, '/Mailman', email.subject + '.txt', oauth_token)
+            formatted_email = FormattedEmail(email)
+            path = '/Mailman/%s' % formatted_email.foldername
```
Comment (rogebrd) [inline backend/mailman_lambda.py:32]: Wouldn't this just set it to the subject not with an extension?

File: backend/mailman_lambda.py [hunk 4]
```diff
@@ -27,7 +28,9 @@ def lambda_handler(event, context):
             )
```
Comment (rogebrd) [inline backend/mailman_lambda.py:28]: Does this logic assume that we will always have a subscription?

**Review by rogebrd — APPROVED** [review]

File: backend/mailman_lambda.py (hunk 4 above)
Comment (mtruong-dbx) [reply backend/mailman_lambda.py:28]: @rogebrd Yes it does. I was planning to go back and add error handling once we had everything working end to end.

File: backend/mailman_lambda.py (hunk 3 above)
Comment (mtruong-dbx) [reply backend/mailman_lambda.py:32]: @rogebrd Yeah, we do this because we need the base path to build the attachment file paths.

**Pull request merged**